### PR TITLE
Add proc macros for Hint and edge

### PR DESCRIPTION
**Pull request by T0mstone**
See also #61

### CODE REVIEW

**Comment by T0mstone** [comment]
To comment on [a5e1249](https://github.com/GraphiteEditor/Graphite/pull/63/commits/a5e1249638c8ab96c0371191efc330a071d02bed): I'm not sure why it suddenly stopped compiling.
What I did was delete the Cargo.lock and let cargo recreate it, which fixed the issue (apart from a slight oversight I fixed with [2f84dd1](https://github.com/GraphiteEditor/Graphite/pull/63/commits/2f84dd1732624be41cb45d532876e9a9a959e197).)

**Comment by TrueDoctor** [comment]
How hard would it be to write some tests?
I unfortunately won't be able to fully review this until tuesday, but I'll try adding some first comments tomorrow

**Comment by T0mstone** [comment]
> How hard would it be to write some tests?

I'm not sure. Tests have to compare the output of functions with a predetermined result, which could turn out tricky with token streams. Doing it with quote should work reasonably well but there may be issues popping up regarding spans.
So if that fails, maybe a `to_string` comparison could work but I'm also not sure on that.

Long story short: I'm not sure and I'll try to write some tests tomorrow and see how it goes

**Review by TrueDoctor — APPROVED** [review]

File: core/proc-macro/src/helpers.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			v.push(x);
+			v
+		}),
+		Err(mut e) => match x {
+			Ok(_) => Err(e),
+			Err(e2) => {
+				e.combine(e2);
+				Err(e)
+			}
+		},
```
[the remark this answers is not in the document]
Comment (T0mstone) [reply core/proc-macro/src/helpers.rs:18]: the wouldn't work too well:
```
Err(mut e) => x.and_then(|_| e)
    .map_err(|e2| { e.combine(e2); e })
```
I think that's way less readable, which is why I chose a `match` expression.

[the remark this answers is not in the document]
Comment (TrueDoctor) [reply core/proc-macro/src/helpers.rs:18]: Yeah alright, was just wondering

**Pull request merged**